File: loadsir/src/main/java/com/hycinth/loadsir/LoadSirUtil.java
package com.hycinth.loadsir;

import android.os.Looper;

import com.hycinth.loadsir.target.ITarget;

import java.util.List;

public class LoadSirUtil {

    public static boolean isMainThread() {
        return Looper.myLooper() == Looper.getMainLooper();
    }

    public static ITarget getTargetContext(Object target, List<ITarget> targetContextList) {
        for (ITarget targetContext : targetContextList) {
            if (targetContext.equals(target)) {
                return targetContext;
            }

        }
        throw new IllegalArgumentException("No TargetContext fit it");
    }
}

File: loadsir/src/main/java/com/hycinth/loadsir/LoadSirUtil.kt
package com.hycinth.loadsir

import android.os.Looper
import com.hycinth.loadsir.target.ITarget

object LoadSirUtil {
    val isMainThread: Boolean = Looper.myLooper() == Looper.getMainLooper()

    fun getTargetContext(target: Any, targetContextList: List<ITarget>): ITarget {
        for (targetContext in targetContextList) {
            if (targetContext == target) {
                return targetContext
            }
        }
        throw IllegalArgumentException("No TargetContext fit it")
    }
}